src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject.cs

```diff
@@ -1009,7 +1024,12 @@ public virtual void Disconnect()
 
                 ClearAnalyzerRuleSet();
 
+                // reinstate pushing down to workspace, so the workspace project remove event fires
```

Comment:
It might have been slightly clearer to make the logic just to bail early if _pushingChangesToWorkspaceHosts was already false, since then you could just be easily toggling it back and forth.